gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/AssignmentColumnHeaderPanel.java

```diff
@@ -369,4 +377,24 @@ private Integer calculateCurrentCategorizedSortOrder(final Long assignmentId) {
 
 		return order;
 	}
+
+	private boolean canUserEditAssignment(GbRole role, Assignment assignment) {
+		boolean canEdit = false;
+		if (role == GbRole.INSTRUCTOR) {
+			canEdit = true;
+		} else {
+			List<PermissionDefinition> permissions = this.businessService.getPermissionsForUser(
```

Comment:
This can possibly be centralised into the BasePage so at least the call is reduced to one. Each page can see the stuff in the basepage so it can just pass it into the panel through the model.